### PR TITLE
[extension/solarwindsapmsettingsextension] Overall structure of the new component 

### PR DESCRIPTION
**Pull request by jerrytfleung**
**Description:** <Describe what has changed.>

- Adding the overall structure of the new component `extension/solarwindsapmsettingsextension` for the first PR

**Link to tracking Issue:** <Issue number if applicable>
[27668](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27668)

**Testing:** <Describe what testing was performed and which tests were added.>
- Added config test
- Added basic factory test and extension test

**Documentation:** <Describe the documentation added.>
- Updated the README.md

### CODE REVIEW

**Comment by jerrytfleung** [comment]
Run the [linting expectations](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components) and it is looking good. Can project maintainer help to approve to rerun workflows?

[linting_expectation.log](https://github.com/open-telemetry/opentelemetry-collector-contrib/files/13813168/linting_expectation.log)

File: extension/solarwindsapmsettingsextension/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+type: solarwindsapmsettings
+
+status:
+  class: extension
+  stability:
+    development: [extension]
+  distributions: [contrib]
```
Comment (atoulme) [inline extension/solarwindsapmsettingsextension/metadata.yaml:7]:
```suggestion
  distributions: []
```

Comment (jerrytfleung) [reply extension/solarwindsapmsettingsextension/metadata.yaml]: Committed the suggested change, regenerated the READMD.md and I also added a yaml file under `.chloggen` so as to meet changelog github action requirement.

**Review by atoulme — APPROVED** [review]

**Comment by jerrytfleung** [comment]
@atoulme, thanks for approving the PR! I noticed there is a failing checks `build-and-test / checks (pull_request)`. I will regenerate the `contrib.yaml` file using `make gendistributions` and update the PR.

**Comment by jerrytfleung** [comment]
> @atoulme, thanks for approving the PR! I noticed there is a failing checks `build-and-test / checks (pull_request)`. I will regenerate the `contrib.yaml` file using `make gendistributions` and update the PR.

DONE

**Comment by atoulme** [comment]
This PR does the job of adding the structure of the component, it passes all checks such as adding it to the versions, github codeowners, factory and so on. The issue has a discussion ongoing on whether this component is applicable to the collector: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27668#issuecomment-1884367716

I am going to remove the "ready to merge" label and add "on-hold" until the discussion is resolved.

**Comment by astencel-sumo** [comment]
> The issue has a discussion ongoing on whether this component is applicable to the collector: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27668#issuecomment-1884367716

I believe the discussion is resolved now (at least from my point of view) in favor of the extension being included in the collector. I suppose we can move forward with this PR then! :tada:

**Comment by atoulme** [comment]
Please rebase so we can make sure you're good to merge.

**Comment by jerrytfleung** [comment]
> Please rebase so we can make sure you're good to merge.

Done. Thank you!

**Comment by jerrytfleung** [comment]
> Please rebase so we can make sure you're good to merge.

Rebased for latest change and it is done

**Comment by jerrytfleung** [comment]
@atoulme Rebased again.  Please help to take a look. Thanks!

**Review by atoulme — APPROVED** [review]
Looks good to me

**Comment by jerrytfleung** [comment]
> Looks good to me

Thanks! Can we merge this PR?

**Pull request merged**